main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
```

Comment:
Sorry, I *think* I meant the function `preprocessLinterPanic`, but that's been changed now anyway.

I continue to think we should ignore the actual error code, it'd remove a lot of lines, and doesn't provide us much value other than a performance improvement by not checking output as often. But it involves more conditionals, and the spec doesn't guarantee a panic will result in an error code 2.

But I'll leave that with you, this looks good to me (and sorry for commenting, let me know if it's not helpful).